### ISSUE TITLE
[BUG] Consume value with `_ = x^`

### ISSUE THREAD

**Issue by rd4com** [modularml/mojo#3435]
### Bug description

Hello, i was writing code to learn the new field sensitivity feature of `nightly 3005`.
A bug seemed to emerge after a while, here is a minimal repro:
```mojo
fn main():
    x = List[Int](1 , 2 , 3)
    _=x^
    _=x^
    List[Int].__del__(x^) #ok consume
    # _=x^ #ok already consumed
```

### Steps to reproduce

Nothing special, all is in the description

### System information

```shell
- `Ubuntu 23.10`
- `mojo 2024.8.3005 (12110d9c)` (nightly)
- `modular 0.7.2 (d0adc668)`
```

**Comment by helehex** [comment]
I've also noticed some other discrepancies with using `_ = x^` vs. `__del__(x^)` before the field sensitivity was added.
Sometimes one would cause a segfault, but not the other, but I couldn't figure out what the difference was.
Seems like the issue has propagated further, as you can now use the value after it's `__del__()` method has been called.

**Comment by lattner** [comment]
This is actually due to some unrelated changes I was working on to improve the model.  The behavior of `x^` has been refined to convert an lvalue to a consumABLE rvalue, but it does not itself consume the value.  as you observe, you can do this with del or other similar methods that take is as an owned value.

**Comment by lattner** [comment]
One thing that `x^` also does fwiw is extend the lifetime of the value.  This was something I ran into because without making "x^" be a use of x, it destroyed the:

```
x = ...
otherstuff() 
_ = x^
```

Pattern.  The natural behavior that falls out is that ^ is a noop, but doing that prevented the pattern above from extending the lifetime of x.  I'm not sure how I feel about it, but I kept the current behavior because it is so pervasive.
